gemma-core/src/main/java/ubic/gemma/persistence/util/QueryUtils.java

```diff
@@ -181,6 +187,70 @@ public static <S extends Comparable<S>, T> Stream<T> streamByBatch( Query query,
                 .flatMap( List::stream );
     }
 
+    /**
+     * Stream the result of a query with the given fetch size.
+     * <p>
+     * This uses offset/limit under the hood because MySQL JDBC does not support scrolling with {@link Query#scroll()}.
+     */
```

Comment:
It would be worth investigating if we can configure MySQL JDBC to actually stream results wiith `scroll()` as that would provide a consistent view of the results being traversed.